overlord/hookstate/context.go

```diff
@@ -137,6 +137,24 @@ func (c *Context) writing() {
 	}
 }
 
+func (c *Context) InitContextData(m map[string]interface{}) error {
+	serialized, err := json.Marshal(m)
+	if err != nil {
+		return err
+	}
+	var data map[string]*json.RawMessage
+	if err := json.Unmarshal(serialized, &data); err != nil {
+		return err
+	}
+
+	if c.IsEphemeral() {
+		c.cache["ephemeral-context"] = data
+	} else {
+		c.task.Set("hook-context", &data)
```

Comment:
is this path used? it doesn't seem so

Reply:
Indeed, thanks! I unexported this.

Reply:
yes, rename and error seems better to me atm

Reply:
mmh, I can see a couple of ways of dealing with this and the XXX ugly comment